pkg/build/controller/strategy/sti.go

```diff
@@ -48,6 +52,11 @@ func (bs *SourceBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod,
 		{Name: "BUILD_LOGLEVEL", Value: fmt.Sprintf("%d", cmdutil.GetLogLevel())},
 	}
 
+	// check if can run container as root
+	if !bs.canRunAsRoot(build) {
+		containerEnv = append(containerEnv, kapi.EnvVar{Name: "ALLOWED_UIDS", Value: "1-"})
+	}
```

Comment:
i'm wary of someone overriding this value with a value specified in strategy.Env.

i know the whitelist protects us today, but if we ever decided to remove the whitelist, we'd be exposed.  maybe it'd be better to set this after merging in the user provided env?